eta/core/features.py

```diff
@@ -383,6 +383,279 @@ def __init__(self, d):
+
+        Attributes:
+            video_path: the input video path
+            frames: an optional frames string to specify the frames of the
+                video to featurize. By default, the value provided in the
+                VideoFramesFeaturizerConfig is used
+            returnX: whether to return the frames matrix
+
+        Returns:
+            If returnX is True, a (# frames) x (# dims) array is returned
+                whose rows contain the computed features
+        '''
+        self._backing_manager(video_path)
+        self.start(warn_on_restart=False, keep_alive=False)
+        v = self._featurize(video_path, returnX)
+        if self._keep_alive is False:
+            self.stop()
+
+        self._backing_manager(video_path, False)
+
+        return v
+
+    def _featurize(self, video_path, returnX=True):
+        if returnX:
+            X = None
+
+        if not self._frame_featurizer:
+            self._frame_featurizer = \
+                self.config.frame_featurizer.build()
+            self._frame_featurizer.start()
+
+        if self._frame_preprocessor is not None:
+            # Pre-process and then featurize the frame
+            v = self._frame_featurizer.featurize(video_path)
+        else:
+            # Featurize the frame
+            v = self._frame_featurizer.featurize(video_path)
+
+        # Write the feature to disk
+        # np.savez_compressed(path, v=v)
+
+        if returnX:
+            if X is None:
+                # Lazily build the GrowableArray now that we know the
+                # dimension of the features
+                X = GrowableArray(len(v))
+            X.update(v)
+
+        if self._frame_featurizer and not self._keep_alive:
+            # Stop the frame featurizer
+            self._frame_featurizer.stop()
+            self._frame_featurizer = None
+
+        return X.finalize() if returnX else None
+
+    def featurized_frame_path(self, frame_number):
+        '''Returns the backing path for the given frame number.'''
+        return os.path.join(
+            self._backing_path, self._frame_string % frame_number)
+
+    def flush_backing(self):
+        '''Deletes all existing feautres on disk in the current backing path.
+        The backing directory itself is not deleted.
+        '''
+        files = [
+            f for f in os.listdir(self._backing_path) if f.endswith(".npz")]
+        for f in files:
+            os.remove(os.path.join(self._backing_path, f))
+
+    def _stop(self):
+        if self._frame_featurizer:
+            self._frame_featurizer.stop()
+            self._frame_featurizer = None
+
+    def update_backing_path(self, backing_path):
+        '''Update the backing path and create the directory tree, if needed.'''
+        self._backing_path = backing_path
+        try:
+            os.makedirs(self._backing_path)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+
+
+class ORBFeaturizer(Featurizer):
```

Comment:
now there are two definitions of `ORBFeaturizer` in this file!!!!!!

Reply:
gonna remove this from this PR, shouldn't be here